File: src/main/java/kavi/tech/service/http/user/ListHandler.java
package kavi.tech.service.http.user;

import io.vertx.core.http.HttpMethod;
import io.vertx.ext.web.RoutingContext;
import kavi.tech.service.service.UserService;
import org.springframework.beans.factory.annotation.Autowired;
import tech.kavi.vs.web.ControllerHandler;
import tech.kavi.vs.web.HandlerRequest;

@HandlerRequest(path = "/list", method = HttpMethod.GET)
public class ListHandler extends ControllerHandler {

    @Autowired
    private UserService userService;

    @Override
    public void handle(RoutingContext routingContext){
        userService.listAll().subscribe(it -> {
            routingContext.response().end(it.toString());
        }, e-> routingContext.response().end(e.getMessage()));
    }
}

File: src/main/kotlin/kavi/tech/service/web/admin/record/ListHandler.kt
package kavi.tech.service.web.admin.record

import io.vertx.core.http.HttpMethod
import io.vertx.core.json.DecodeException
import io.vertx.core.json.JsonObject
import io.vertx.ext.mongo.FindOptions
import io.vertx.ext.web.Route
import io.vertx.ext.web.RoutingContext
import org.springframework.beans.factory.annotation.Autowired
import tech.kavi.vs.web.ControllerHandler
import tech.kavi.vs.web.HandlerRequest
import io.vertx.ext.web.api.validation.HTTPRequestValidationHandler
import io.vertx.ext.web.api.validation.ParameterType
import io.vertx.ext.web.handler.impl.HttpStatusException
import kavi.tech.service.common.extension.success
import kavi.tech.service.common.extension.toEntity
import kavi.tech.service.mongo.model.RecordModel
import kavi.tech.service.mysql.dao.UserDao
import kavi.tech.service.mysql.entity.User

@HandlerRequest(path = "/list", method = HttpMethod.GET)
class ListHandler @Autowired constructor(
        private val recordModel: RecordModel
) : ControllerHandler() {

    /**
     * 数据处理
     * */
    override fun handle(routingContext: RoutingContext) {
        recordModel.list(JsonObject(), FindOptions())
            .subscribe({
                routingContext.response().success(it)
            }, routingContext::fail)
    }
}